src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Components/ResultHelper.cs

```diff
@@ -0,0 +1,114 @@
+    /// <param name="searchControllerResults">List with all search controller matches</param>
+    /// <param name="icon">The path to the result icon</param>
+    /// <returns>List of results</returns>
+    internal static List<WindowWalkerListItem> GetResultList(List<SearchResult> searchControllerResults, bool isKeywordSearch, string icon, string infoIcon)
+    {
+        if (searchControllerResults == null || searchControllerResults.Count == 0)
+        {
+            return new List<WindowWalkerListItem>();
+        }
+
+        List<WindowWalkerListItem> resultsList = new List<WindowWalkerListItem>(searchControllerResults.Count);
+        var addExplorerInfo = searchControllerResults.Any(x =>
+            string.Equals(x.Result.Process.Name, "explorer.exe", StringComparison.OrdinalIgnoreCase) &&
+            x.Result.Process.IsShellProcess);
+
+        // Process each SearchResult to convert it into a Result.
+        // Using parallel processing if the operation is CPU-bound and the list is large.
+        resultsList = searchControllerResults
+            .AsParallel()
+            .Select(x => CreateResultFromSearchResult(x, icon))
+            .ToList();
+
+        if (addExplorerInfo && isKeywordSearch && !SettingsManager.Instance.HideExplorerSettingInfo)
+        {
+            resultsList.Add(GetExplorerInfoResult(infoIcon));
+        }
+
+        return resultsList;
+    }
+
+    /// <summary>
+    /// Creates a Result object from a given SearchResult.
+    /// </summary>
+    /// <param name="searchResult">The SearchResult object to convert.</param>
+    /// <param name="icon">The path to the icon that should be used for the Result.</param>
+    /// <returns>A Result object populated with data from the SearchResult.</returns>
+    private static WindowWalkerListItem CreateResultFromSearchResult(SearchResult searchResult, string icon)
+    {
+        var item = new WindowWalkerListItem(searchResult.Result)
+        {
+            Title = searchResult.Result.Title,
+            Icon = new(icon),
+            Subtitle = GetSubtitle(searchResult.Result),
+        };
+        item.MoreCommands = ContextMenuHelper.GetContextMenuResults(item).ToArray();
+
+        return item;
+    }
+
+    /// <summary>
+    /// Returns the subtitle for a result
+    /// </summary>
+    /// <param name="window">The window properties of the result</param>
+    /// <returns>String with the subtitle</returns>
+    private static string GetSubtitle(Window window)
+    {
+        if (window == null || !(window is Window))
+        {
+            return string.Empty;
+        }
+
+        var subtitleText = Resources.wox_plugin_windowwalker_Running + ": " + window.Process.Name;
+
+        if (SettingsManager.Instance.SubtitleShowPid)
+        {
+            subtitleText += $" ({window.Process.ProcessID})";
+        }
+
+        if (!window.Process.IsResponding)
+        {
+            subtitleText += $" [{Resources.wox_plugin_windowwalker_NotResponding}]";
```

Comment:
feels like a tag to me

Reply:
You know, it's probably going to be a great exercise to run through every single flow we have out of the box once the real app is ready and just rethink some of the dynamics. I think tag feels right too here.